lib/zstd.h

```diff
@@ -578,7 +579,8 @@ ZSTDLIB_API size_t ZSTD_CCtx_reset(ZSTD_CCtx* cctx, ZSTD_ResetDirective reset);
  *  Should cctx hold data from a previously unfinished frame, everything about it is forgotten.
  *  - Compression parameters are pushed into CCtx before starting compression, using ZSTD_CCtx_set*()
  *  - The function is always blocking, returns when compression is completed.
- *  Hint : compression runs faster if `dstCapacity` >=  `ZSTD_compressBound(srcSize)`.
+ *  NOTE: Providing `dstCapacity >= ZSTD_compressBound(srcSize)` guarantees that zstd will have
+ *        enough space to successfully compress the data, though it is possible it fails for other reasons.
```

Comment:
> though it is possible it fails for other reasons

What other reasons do you have in mind ?
Memory allocation failures ?
Providing `dstCapacity >= ZSTD_compressBound(srcSize)` should be enough to avoid #3524.

Reply:
I mean `ZSTD_compress2()` can let you specify a lot of options. E.g. compression could fail to allocate a thread. We could run out of memory.

I only mentioned this in `ZSTD_compress2()`, and skipped it in `ZSTD_compress()`, but I just mean that you still have to check the error code.